src/api/mod.rs

```diff
@@ -32,73 +31,96 @@ pub async fn api_wrapper<Response, Form>(
-           .send()
-           .await?
-           .text()
-           .await?
-       }
-       HttpType::Post => client.post(path).json(form).send().await?.text().await?,
-       HttpType::Put => client.put(path).json(form).send().await?.text().await?,
-     };
-   } else {
-     use wasm_bindgen::UnwrapThrowExt;
-
-     let abort_controller = web_sys::AbortController::new().ok();
-     let abort_signal = abort_controller.as_ref().map(|a| a.signal());
-
-     json = match type_ {
-       HttpType::Get => {
-         gloo_net::http::Request::get(&build_fetch_query(route, form))
-           .abort_signal(abort_signal.as_ref())
-           .send()
-           .await?
-           .text()
-           .await?
-       }
-       HttpType::Post => {
-         gloo_net::http::Request::post(route)
-           .json(form)
-           .expect_throw("Could not parse json body")
-           .abort_signal(abort_signal.as_ref())
-           .send()
-           .await?
-           .text()
-           .await?
-       }
-       HttpType::Put => {
-         gloo_net::http::Request::put(route)
-           .json(form)
-           .expect_throw("Could not parse json body")
-           .abort_signal(abort_signal.as_ref())
-           .send()
-           .await?
-           .text()
-           .await?
-       }
-     };
-
-     // abort in-flight requests if the Scope is disposed
-     // i.e., if we've navigated away from this page
-     leptos::on_cleanup( move || {
-       if let Some(abort_controller) = abort_controller {
-         abort_controller.abort()
-       }
-     });
-   }
+      let client = reqwest::Client::new();
+
+      let mut request_builder = match type_ {
+        HttpType::Get => client.get(&build_fetch_query(route, form)),
+        HttpType::Post => client.post(route),
+        HttpType::Put => client.put(route),
+      };
+
+      match get_cookie_wrapper("jwt").await {
+        Ok(jwt) => {
+          if jwt.len() > 0 {
+            request_builder = request_builder.header("Authorization", &format!("Bearer {}", jwt)[..]);
+          }
+        },
+        Err(_) => {
+        },
+      };
+
+      json = match type_ {
+        HttpType::Get => request_builder.send().await?.text().await?,
+        HttpType::Post => request_builder.json(form).send().await?.text().await?,
+        HttpType::Put => request_builder.json(form).send().await?.text().await?,
+      };
+    } else {
+      use wasm_bindgen::UnwrapThrowExt;
+
+      let abort_controller = web_sys::AbortController::new().ok();
+      let abort_signal = abort_controller.as_ref().map(|a| a.signal());
+
+      let mut request_builder = match type_ {
+        HttpType::Get => gloo_net::http::Request::get(&build_fetch_query(route, form)),
+        HttpType::Post => gloo_net::http::Request::post(route),
+        HttpType::Put => gloo_net::http::Request::put(route),
+      };
+
+      match get_cookie_wrapper("jwt").await {
+        Ok(jwt) => {
+          if jwt.len() > 0 {
```

Comment:
This length check is probably pointless. If that cookie exists, then we should be fine.

Reply:
agreed